body.tex

```diff
@@ -0,0 +1,440 @@
+
+The following elements should not be part of a software product:
+
+\begin{itemize}
+\item build tools, otherwise we will have build tools deployed in operations.
+\item environment definition, since the environment depends on the final instantiation of the software product that are not visible in the development phase.
+\end{itemize}
+
+These elements shall follow a parallel development / release process.
+
+Test data should also not be included in a software product. However including small datasets for enabling the unit tests is not a bad practice.
+
+The content of this subsection is a \textit{prerequisite} for any release procedure to be applied.
+
+
+\subsection{Dependencies} \label{sec:dependencies}
+
+A package may only depend on other packages. Third party dependencies are either packages or part of the conda environment.
+
+The dependency information is provided in each Git package in specific dependency files  according to the build system e.g.  requirements.txt, EUPS table file, etc.
+These files , and hence the dependencies,  are considered part of the source code and controlled  and versioned in the same manner.
+
+
+
+
+\subsection{Software Release} \label{sec:swrel}
+
+A software release is a consciously identified tag of a software product repository documented with a software release note.
+The identifier is usually of the form \texttt{M.n.p}.
+
+The tag in the Github repository and the software release note is sufficient to identify the release, and therefore for a developer
+to resolve the dependencies, build the binaries and execute the software.
+
+The software release is a monolithic snapshot of the software product to be used as is, or in its corresponding binary package(s), by the downstream processes or users.
+In theory, it should be possible to identify the release with a checksum calculated automatically on the delivered package.
+That checksum can be used to identify the release installed in an environment, or included in a distribution.
+
+
+\subsection{Binary Packages} \label{sec:swbpkg}
+
+A binary package is a package containing executable binaries for the corresponding release.
+It is created by building the SW provided in the release tag.
+Binary packages can be created to support multiple platforms (such as Linux, OSX, windows) if required.
+
+In some cases, these packages do not contain compiled binaries, but executable scripts or just text files.
+
+Binaries should be generated only once, and made available for their use by dependent software or for deployment.
+
+The majority of DM software is using EUPS for binary packaging. Other platforms can be considered, for example \textit{conda} or \textit{pypi}.
+Software products implemented in java are using jar/war binary packaging.
+\footnote{So far there are a few technologies used to handle binary packages. It is recommended to assess and optimize them, converging to use only one.
+Conda seems to have become a standard for python based projects.}
```

Comment:
seems...

Reply:
Well, it doesn't seem so to me. :-)

Seems like this text has been commented in current `master`. In general, please remove rather than commenting text — Git won't forget it, so you lose nothing, and it makes the source much easier to read.

Reply:
I remove the commented text.